daemon/cmd/daemon.go

```diff
@@ -690,7 +690,7 @@ func NewDaemon(ctx context.Context, epMgr *endpointmanager.EndpointManager, dp d
 	}
 
 	// Must occur after d.allocateIPs(), see GH-14245 and its fix.
-	d.nodeDiscovery.StartDiscovery(nodeTypes.GetName())
+	d.nodeDiscovery.StartDiscovery()
```

Comment:
Could we backport 9c1cb4856322a80174799dae142ef761e51e2310 instead of cherry-picking a subset of its changes?